brokerapi/brokers/gcp_service_broker.go

```diff
@@ -396,18 +396,30 @@ func (gcpBroker *GCPServiceBroker) LastOperation(instanceID string) (models.Last
 		return models.LastOperation{}, models.ErrInstanceDoesNotExist
 	}
 
-	if gcpBroker.ServiceBrokerMap[instance.ServiceId].Async() {
+	if gcpBroker.ServiceBrokerMap[instance.ServiceId].ProvisionsAsync() || gcpBroker.ServiceBrokerMap[instance.ServiceId].DeprovisionsAsync() {
 		done, err := gcpBroker.ServiceBrokerMap[instance.ServiceId].PollInstance(instanceID)
 		if err != nil {
 			if gerr, ok := err.(*googleapi.Error); ok {
 				if gerr.Code == 503 {
 					return models.LastOperation{State: models.InProgress, Description: ""}, err
 				}
 			}
+			// This is not a retryable error. Return fail
 			return models.LastOperation{State: models.Failed, Description: ""}, err
 		} else {
 
 			if done {
+				// no error and we're done! Delete from the SB database if this was a delete flow and return success
+				deleteFlow, err := gcpBroker.ServiceBrokerMap[instance.ServiceId].LastOperationWasDelete(instanceID)
+				if err != nil {
+					return models.LastOperation{State: models.Succeeded, Description: ""}, fmt.Errorf("Couldn't determine if provision or deprovision flow, this may leave orphaned resources, contact your operator for cleanup")
+				}
+				if deleteFlow {
+					err = db_service.SoftDeleteInstanceDetails(instanceID)
```

Comment:
did you consider moving this into `PoolInstance`? That way you don't need to expose `LastOperationWasDelete`